internal/controller/companion_controller.go

```diff
@@ -130,3 +165,63 @@ func (r *Reconciler) loggerWithCompanion(companion *kcmv1alpha1.Companion) *zap.
 		"name", companion.GetName(),
 	)
 }
+
+func (r *Reconciler) reconcileDeployment(ctx context.Context, companion *kcmv1alpha1.Companion,
+	log *zap.SugaredLogger,
+) error {
+	// define deployment object.
+	expectedDeployment, err := r.backendManager.GenerateNewDeployment(companion, r.config.KymaCompanionBackendImage)
+	if err != nil {
+		return err
+	}
+
+	// fetch existing deployment.
+	existingDeployment, err := r.kubeClient.GetDeployment(ctx, expectedDeployment.GetName(),
+		expectedDeployment.GetNamespace())
+	if err != nil && !kapierrors.IsNotFound(err) {
+		return err
+	}
+
+	// compare if the deployment needs to be updated.
+	if equality.Semantic.DeepEqual(existingDeployment, expectedDeployment) {
```

Comment:
if the deployment doesn't exist yet, this will simply evaluate to `false` and the deployment is applied in the updating step below?

Reply:
yes, `nil` will be evaluated to `false`.